### PR TITLE
docs: add wechat SDK manual

### PR DESCRIPTION
**Pull request by jingnanl**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

Add WeChat SDK manual

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Review by zxkane — CHANGES_REQUESTED** [review]
Let's prioritize the Chinese version.

**Review by zxkane — APPROVED** [review]

File: docs/en/sdk-manual/wechat.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
+# Clickstream WeChat Miniprogram SDK
```
Comment (zhu-xiaowei) [inline docs/en/sdk-manual/wechat.md:1]: Typo: In word 'Miniprogram' , refer to https://developers.weixin.qq.com/miniprogram/en/dev/framework/

File: docs/en/sdk-manual/wechat.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+In Clickstream WeChat Miniprogram SDK, we do not set duration max limit for a session. If the user opens the mini program within a certain time period after they quit the mini program, we still consider it being in the same session.
+
+1. **_session_start**: When the mini program is launched for the first time, or the mini program is open to the foreground and the time between the last exit exceeds `session_time_out` period.
+
+1. **_session_duration**: We calculate the `_session_duration` based on the current event creation timestamp and the session's `_session_start_timestamp`, this attribute will be added in every event during the session.
+
+1. **session_time_out**: 30 minutes by default, which can be customized via the configuration API.
+
+1. **_session_number**: The total number of distinct sessions (by session id), this attribute will be added in every event's attribute object.
+
```
Comment (zhu-xiaowei) [inline docs/en/sdk-manual/wechat.md:182]: Item is not correctly numbered.

File: docs/zh/sdk-manual/wechat.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+
+// 用户退出登录后清空用户ID
+ClickstreamAnalytics.setUserId(null);
+```
+
+当前登录用户的属性会缓存在小程序本地存储空间（wxStorage）中。
+
+#### 记录事件
+
+您可以调用`ClickstreamAnalytics.record()`方法记录自定义事件。事件中的参数`name`是必需的，参数`attributes`和`items`不是必需的。参数`attributes`是一个对象，用来记录各种属性。参数`items`是一个保存`item`类型对象的数组，请参阅[item属性](#_17)查看item中的参数。
```
Comment (zhu-xiaowei) [inline docs/zh/sdk-manual/wechat.md:80]: `[item属性](#_17)` -> `[item属性](#item) `

File: docs/zh/sdk-manual/wechat.md [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+  "system_language": "en",
+  "screen_height": 667,
+  "screen_width": 375,
+  "zone_offset": 28800000,
+  "network_type": "wifi",
+  "sdk_version": "0.2.0",
+  "sdk_name": "aws-solution-clickstream-sdk",
+  "app_version": "1.0.1",
+  "app_package_name": "wxbd614036ba3d1f05",
+  "user": {
```
Comment (zhu-xiaowei) [inline docs/zh/sdk-manual/wechat.md:283]: Let's add an item example to show the structure of the items array

File: docs/en/sdk-manual/wechat.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+  "system_language": "en",
+  "screen_height": 667,
+  "screen_width": 375,
+  "zone_offset": 28800000,
+  "network_type": "wifi",
+  "sdk_version": "0.2.0",
+  "sdk_name": "aws-solution-clickstream-sdk",
+  "app_version": "1.0.1",
+  "app_package_name": "wxbd614036ba3d1f05",
+  "user": {
```
[the remark this answers is not in the document]
Comment (jingnanl) [reply docs/en/sdk-manual/wechat.md:289]: done

File: docs/en/sdk-manual/wechat.md (hunk 1 above)
Comment (jingnanl) [reply docs/en/sdk-manual/wechat.md:1]: Synced offline. We'll use "WeChat **Miniprogram** SDK" as the SDK name and use "WeChat Mini Program" in the docs and code comments.

File: docs/zh/sdk-manual/wechat.md (hunk 3 above)
Comment (jingnanl) [reply docs/zh/sdk-manual/wechat.md:80]: done

File: docs/en/sdk-manual/wechat.md (hunk 2 above)
Comment (jingnanl) [reply docs/en/sdk-manual/wechat.md:182]: It works in Markdown and it is rendered correctly in local.

File: docs/zh/sdk-manual/wechat.md (hunk 4 above)
Comment (jingnanl) [reply docs/zh/sdk-manual/wechat.md:283]: done

**Review by zhu-xiaowei — APPROVED** [review]

**Pull request merged**